src/main/java/com/iopipe/SystemMeasurement.java

```diff
@@ -232,16 +321,19 @@ private static String __readFirstLine(Path __p, String __def)
 		if (__p == null)
 			throw new NullPointerException();
 		
-		try
+		try (BufferedReader br = new BufferedReader(new InputStreamReader(
+			Files.newInputStream(__p, StandardOpenOption.READ))))
 		{
-			for (String l : Files.readAllLines(__p))
+			for (;;)
```

Comment:
infinite loop?

Reply:
This loop is not an infinite loop, it will return the default value when EOF is reached or the first non-blank line was found.

Reply:
The returns are both in `if`s so this could be infinite

Reply:
wondering what the gain was from changing this from having a clear limit?

Reply:
The loop will only be infinite if the file contains an infinite amount of lines.

Before: Read all lines from the file then return those lines, iterating through each one and finding the first non-blank line.
Now: Read the file line by line, for each line find the first non-blank line.

Since we only care about the first line we see in the file, we do not need to read the entire file just to get to the first line. So this makes things faster by removing wasted reads of a file that we will pretty much just be discarding anyway.

Reply:
got it, thanks!